posixlib/src/main/resources/scala-native/types.h

```diff
@@ -10,6 +10,7 @@ typedef unsigned int scalanative_uid_t;
 typedef unsigned int scalanative_gid_t;
 typedef long long scalanative_off_t;
 typedef long int scalanative_time_t;
+typedef struct timespec scalanative_timespec;
```

Comment:
I believe this line is improper.  POSIX 2018 (and many before, I believe) specify `timespec` as being defined
in `time.h`, not `types.h` ( probably `time` was well established before `types.h` was specified.)

Some else in this PR there is probably an `import` which
needs to change or happen.

Reply:
Thank you for feedback! I will address this later.

Reply:
Hmm, could you tell me why `scalanative_timespec` introduced here? If we use timespec from time.h, I don't think scalanative_timespec is needed.

https://github.com/scala-native/scala-native/pull/2713/files#diff-9db9f7da0930f16af30649420a8fd8857d1b9f1e07366491539a6d5ef0c0eeccR36

Reading the comment in sys/stat.c, I guess src/main/resources/scala-native/types.h intentionally violates spec for portability.

```
// We don't use the "standard" types such as `dev_t` for instance
// because these have different sizes on eg. Linux and OSX. We use the
// smallest type that can hold all the possible values for the different
// systems.
struct scalanative_stat {
    scalanative_dev_t st_dev;     /** Device ID of device containing file. */
    scalanative_dev_t st_rdev;    /** Device ID (if file is character or block
                                      special). */
    scalanative_ino_t st_ino;     /** File serial number. */
```

These types above were introduced by Martin Duhem at 2017. Is this statement "these have different sizes on eg. Linux and OSX" not the case for recent versions?

Reply:
NOTE: I'm not in a hurry😉 Have a nice holiday!

Reply:
i10416 You ask good questions.   Let me finish my 
highest level concern, so the PR can progress, and return to the trickery at play here.

Reply:
Quick answer to 1st question.  I'll get back to second later (tomorrow?)

Lines 92-102 check that Scala Natives scala timespec can be passed thru
to C with no "copy-in/copy-out" code.
```
// struct timespec
_Static_assert(sizeof(struct scalanative_timespec) == sizeof(struct timespec),
               "Unexpected size: struct timespec");

_Static_assert(offsetof(struct scalanative_timespec, tv_sec) ==
                   offsetof(struct timespec, tv_sec),
               "offset mismatch: timespec.tv_sec");
// and so on.
```
From software engineering view, it is unfortunate that the C declaration of scalanative_timespec
is hand edited and hand synchronized with the corresponding Scala struct. The test would
be stronger if the former where generated from the latter.  For now, the costly safeguards
are education & review.  In someplaces there are "If you change .scala, also change .c" comments.

Reply:
Thank you!
So, 
1. there exist `scalanative_timespec` and other `scalanative_*` structs so that we can verify Scala struct is the same format as c struct.

>From software engineering view, it is unfortunate that the C declaration of scalanative_timespec
is hand edited and hand synchronized with the corresponding Scala struct. The test would
be stronger if the former where generated from the latter. For now, the costly safeguards
are education & review. In someplaces there are "If you change .scala, also change .c" comments.

2. I guess you mean it is better to put `typedef struct timespec scalanative_timespec;` in `time.h` for future maintenance, though original PR #1610 puts it in `types.h`.

Am I correct?